src/main/java/Duke.java

```diff
@@ -1,10 +1,283 @@
+                System.out.println("____________________________________________________________");
+            }
+            System.out.println("____________________________________________________________");
+            command = scanner.nextLine();
+        }
+        duke.saveTasksToFile(tasks);
+    
+        System.out.println("____________________________________________________________");
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println("____________________________________________________________");
+    }
+    
+    
+    private ArrayList<Task> loadTasksFromFile() {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            File file = new File(FILE_PATH);
+            if (!file.exists()) {
+                file.createNewFile();
+            } else {
+                Scanner scanner = new Scanner(file);
+                while (scanner.hasNextLine()) {
+                    String line = scanner.nextLine();
+                    Task task = parseTaskFromString(line);
+                    tasks.add(task);
+                }
+                scanner.close();
+            }
+        } catch (IOException e) {
+            System.out.println("Error loading tasks from file: " + e.getMessage());
+        }
+        return tasks;
+    }
+
+    private Task parseTaskFromString(String line) {
+        String[] parts = line.split("\\|");
+        String type = parts[0].trim();
+        boolean isDone = parts[1].trim().equals("1"); // Assuming "1" represents task is done
+        String description = parts[2].trim();
+    
+        switch (type) {
+            case "D":
+                String by = parts[3].trim();
+                LocalTime byTime = LocalTime.parse(by, DateTimeFormatter.ofPattern("HHmm"));
+                return new DeadlineTask(description, byTime);
+            case "E":
+                String from = parts[3].trim();
+                String to = parts[4].trim();
+                LocalTime startTime = LocalTime.parse(from, DateTimeFormatter.ofPattern("HHmm"));
+                LocalTime endTime = LocalTime.parse(to, DateTimeFormatter.ofPattern("HHmm"));
+                return new EventTask(description, startTime, endTime);
+            default:
+                return new ToDoTask(description);
+        }
+    }
+    
+    
+    
+
+    private  void saveTasksToFile(ArrayList<Task> tasks) {
+        try (PrintWriter writer = new PrintWriter(FILE_PATH)) {
+            for (Task task : tasks) {
+                writer.println(task.toFileString());
+            }
+        } catch (IOException e) {
+            System.out.println("Error saving tasks to file: " + e.getMessage());
+        }
+    }
+}
+
+// Task, ToDoTask, DeadlineTask, EventTask classes remain unchanged
+
+class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
```

Comment:
Is there a reason why you added this. infront in task and not in Duke class? Can consider having a constant reference standard.